src/main/java/fi/otavanopisto/kuntaapi/server/index/FileSearcher.java

```diff
@@ -0,0 +1,114 @@
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHitField;
+
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.id.PageId;
+import fi.otavanopisto.kuntaapi.server.id.FileId;
+import fi.otavanopisto.kuntaapi.server.id.IdController;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+
+@ApplicationScoped
+public class FileSearcher {
+  
+  private static final String TYPE = "file";
+  private static final String FILE_ID_FIELD = "fileId";
+  private static final String PAGE_ID_FIELD = "pageId";
+  private static final String ORGANIZATION_ID_FIELD = "organizationId";
+  
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private IdController idController;
+  
+  @Inject
+  private IndexReader indexReader;
+
+  public SearchResult<FileId> searchFiles(OrganizationId organizationId, PageId pageId, String queryString, Long firstResult, Long maxResults) {
+    OrganizationId kuntaApiOrganizationId = idController.translateOrganizationId(organizationId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiOrganizationId == null) {
+      logger.severe(String.format("Failed to translate organization %s into Kunta API id", organizationId.toString()));
+      return new SearchResult<>(Collections.emptyList());
+    }
+
+    BoolQueryBuilder query = boolQuery()
+        .must(matchQuery(ORGANIZATION_ID_FIELD, kuntaApiOrganizationId.getId()));
+    
+    if (queryString != null) {
+      query.must(queryStringQuery(queryString));
+    }
+
+    if (pageId != null) {
+      PageId kuntaApiPageId = idController.translatePageId(pageId, KuntaApiConsts.IDENTIFIER_NAME);
+      if (kuntaApiPageId == null) {
+        logger.severe(String.format("Failed to translate page %s into Kunta API id", pageId.toString()));
+        return new SearchResult<>(Collections.emptyList());
+      }
+      
+      query.must(matchQuery(PAGE_ID_FIELD, kuntaApiPageId.getId()));
+    }
+    
+    return searchFiles(query, firstResult, maxResults);
+  }
+  
+  private SearchResult<FileId> searchFiles(QueryBuilder queryBuilder, Long firstResult, Long maxResults) {
+    if (!indexReader.isEnabled()) {
+      logger.warning("Could not execute search. Search functions are disabled");
+      return null;
+    }
+    
+    SearchRequestBuilder requestBuilder = indexReader
+      .requestBuilder(TYPE)
+      .storedFields(FILE_ID_FIELD, ORGANIZATION_ID_FIELD)
+      .setQuery(queryBuilder);
+    
+    if (firstResult != null) {
+      requestBuilder.setFrom(firstResult.intValue());
+    }
+    
+    if (maxResults != null) {
+      requestBuilder.setSize(maxResults.intValue());
+    }
+      
+    return new SearchResult<>(getFileIds(indexReader.search(requestBuilder)));
+  }
+  
+  private List<FileId> getFileIds(SearchHit[] hits) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getFileIds" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)